recipes/mcmc-sampling-emcee/mcmc_sampling.ipynb

```diff
@@ -45,7 +45,7 @@
   },
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #4.        for factor, par in zip(pars, dataset.models.parameters.free_parameters):`
</pre><p><code>pars</code> is a <code>factor</code> in fact?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gammapy/gammapy-recipes/pull/32/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Reply:
par was badly chosen, in fact par is a gammapy parameter object and pars is a list of factors.
But it more conventional to have lnprob() function taking list of pars as input 
and the factor thing is an internal gammapy thing.
Not sure what to do 
for now I'm changing to  . What do you think ?

```

def lnprob(pars, dataset):
    """
    Estimate the likelihood of a model including prior on parameters.
    Input : 
    pars : a list of parameters
    dataset: a gammapy dataset 
    """
    # Update model parameters factors inplace
    # The MCMC sampler will evaluate the likelihood of the model given
    # a set of parameters. We need to update the model parameters value before 
    # evaluating the new likelihood value.
    for factor, parameter in zip(pars, dataset.models.parameters.free_parameters):
        parameter.factor = factor # What the sampler will need is the factor (e.g. 1) instead of the value (1e-12)
        
    # dataset.likelihood returns Cash statistics values
    # emcee will maximisise the LogLikelihood so we need -dataset.likelihood
    total_lnprob = -dataset.stat_sum() #stat_sum now includes stat + stat_priors

    return total_lnprob
```

Reply:
<p>thanks</p>